pytorch_lightning/plugins/training_type/sharded.py

```diff
@@ -47,6 +52,12 @@ def _reinit_optimizers_with_oss(self):
             if not isinstance(optimizer, OSS):
                 optim_class = type(optimizer)
                 zero_optimizer = OSS(params=optimizer.param_groups, optim=optim_class, **optimizer.defaults)
+                if _FAIRSCALE_OSS_FP16_BROADCAST_AVAILABLE:
+                    is_fp16 = self.lightning_module.trainer.precision == 16
+                    # For multi-node training, compressing the model shards in fp16 before broadcasting
+                    # improves performance. When using PyTorch AMP, it will not degrade
+                    # the model performance.
+                    zero_optimizer.broadcast_fp16 = is_fp16 and self.num_nodes > 1
```

Comment:
Could we add a test for this? It being True, for 16bit precision and multi node.

Reply:
> Could we add a test for this? It being True, for 16bit precision and multi node.

was wondering do we have multi nodes testing example.

Reply:
It seems to be disabled for now. https://github.com/PyTorchLightning/pytorch-lightning/blob/master/tests/accelerators/test_multi_nodes_gpu.py

cc: @Borda @SeanNaren @tchaton

Reply:
let me know if I should add one if multi-node testing is currently disabled. Maybe i could add in the same file for now (which might be easier for re-enable)

Reply:
Yup, that should work. Thanks!

Reply:
yes, we are in process of adding multi-node back...